File: backend/projectverseAPI/projectverseAPI/Models/CollaborationApplicant.cs
namespace projectverseAPI.Models
{
    public class CollaborationApplicant
    {
        public Guid Id { get; set; }
        public Guid ApplicantUserId { get; set; }
        public User? ApplicantUser { get; set; }
        public Guid AppliedPositionId { get; set; }
        public CollaborationPosition? AppliedPosition { get; set; }
        public Guid AppliedCollaborationId { get; set; }
        public Collaboration? AppliedCollaboration { get; set; }
        public ApplicationStatus ApplicationStatus { get; set; }
        public DateTime AppliedOn { get; set; }
    }

    public enum ApplicationStatus
    {
        Rejected,
        Pending,
        Accepted
    }
}

File: frontend/projectverse/src/data/CollaborationApplicant.ts

import User from "./User"

interface CollaborationApplicant{  
  
  id:string,
  applicantUserName:string,
  applicantEmail:string,
  applicantUserId:string,
  appliedPositionId:string,
  applicationStatus : ApplicationStatus,
  appliedOn:Date
  
}

enum ApplicationStatus
{
    Rejected,
    Pending,
    Accepted
}

export default CollaborationApplicant;